superset/models/core.py

```diff
@@ -652,13 +655,66 @@ def export_dashboards(cls, dashboard_ids):
                 make_transient(eager_datasource)
                 eager_datasources.append(eager_datasource)
 
-        return json.dumps({
-            'dashboards': copied_dashboards,
-            'datasources': eager_datasources,
-        }, cls=utils.DashboardEncoder, indent=4)
+            files = []
+            total_file_size = 0
+            total_file_rows = 0
+            if export_data and export_data_dir:
+
+                for data_table in eager_datasources:
+                    engine = data_table.database.get_sqla_engine()
+                    columns = [c.get_sqla_col() for c in data_table.columns]
+
+                    qry = (
+                        select(columns)
+                        .select_from(text(data_table.name))
+                    )
+                    qry.compile(engine)
+                    sql = '{}'.format(
+                        qry.compile(engine),
+                    )
+
+                    df = pd.read_sql_query(sql=sql, con=engine)
+                    row_count = len(df.index) + 1  # plus one for header
+
+                    file_name = f'{data_table.name}.csv.gz'
+                    file_path = f'{export_data_dir}/{file_name}'
+
+                    if not os.path.exists(export_data_dir):
+                        os.makedirs(export_data_dir)
+                    df.to_csv(file_path, compression='gzip')
+
+                    file_size = os.path.getsize(file_path)
+
+                    table_record = {
+                        'file_name': file_name,
+                        'rows': row_count,
+                        'size': file_size,
+                        'table_name': data_table.name,
+                    }
+
+                    total_file_rows += row_count
+                    total_file_size += file_size
+
+                    files.append(table_record)
+
+            # Partially fill out the bibliography
+            desc = {
+                'total_size': total_file_size,
+                'total_size_mb': round(total_file_size / (1024.0 * 1024.0), 2),
+                'total_rows': total_file_rows,
+                'file_count': len(files),
+                'created_at': datetime.now().isoformat(),
+            }
+
+        return {
+            'description': desc,
+            'dashboards': [o.export_to_json() for o in copied_dashboards],
```

Comment:
Does that mean that we have JSON (as a long string) inside JSON here?

Reply:
No, the name could use improvement. It means serialize-to-json, not actually to_json. It prepares the objects for jsonification. The only place we have json-within-json is params, which I would love to fix at some point if possible.

Reply:
To me `to_json` implies the thing is going to return a jsonified string ala `json.dumps`. Maybe `export_to_json_serializable`

Reply:
Done.